src/test/java/fr/insee/pogues/service/CodesListServiceTest.java

```diff
@@ -0,0 +1,264 @@
+package fr.insee.pogues.service;
+
+import fr.insee.pogues.exception.CodesListException;
+import fr.insee.pogues.model.CodeList;
+import fr.insee.pogues.model.QuestionType;
+import fr.insee.pogues.model.Questionnaire;
+import fr.insee.pogues.persistence.service.QuestionnairesService;
+import fr.insee.pogues.utils.PoguesSerializer;
+import fr.insee.pogues.webservice.model.dtd.codelists.Code;
+import fr.insee.pogues.webservice.model.dtd.codelists.CodesList;
+import fr.insee.pogues.webservice.model.dtd.codelists.ExtendedCodesList;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.util.List;
+import java.util.Objects;
+
+import static fr.insee.pogues.utils.ModelCreatorUtils.initFakeCodeLists;
+import static fr.insee.pogues.utils.Utils.findQuestionWithId;
+import static fr.insee.pogues.utils.Utils.loadQuestionnaireFromResources;
+import static fr.insee.pogues.utils.model.CodesList.getListOfQuestionIdWhereCodesListIsUsed;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.MockitoAnnotations.initMocks;
+
+class CodesListServiceTest {
+
+    @Mock
+    QuestionnairesService questionnairesService;
+
+    @InjectMocks
+    private CodesListService codesListService;
+
+    @BeforeEach
+    void initQuestionnaireService(){
+        codesListService = new CodesListService(questionnairesService);
+        initMocks(this);
+    }
+
```

Comment:
- `initMocks` : deprecated -> use `@ExtendWith(MockitoExtension.class)`
- @InjectMocks ne sert à rien ici car tu injectes déjà le mock comme dépendance dans le `BeforeEach`
```suggestion
@ExtendWith(MockitoExtension.class)
class CodesListServiceTest {

    @Mock
    QuestionnairesService questionnairesService;

    private CodesListService codesListService;

    @BeforeEach
    void initQuestionnaireService(){
        codesListService = new CodesListService(questionnairesService);
    }
    
```